docs/2022-isc22/troubleshooting.md

```diff
@@ -1 +1,485 @@
+
+Can you fix the problem you run into, perhaps without even changing
+the easyconfig file?
+
+??? success "(click to show solution)"
+
+    The installation fails because the source file `subread-2.0.1-source.tar.gz`
+    is not found:
+    ```
+    $ eb subread.eb
+    ...
+    == FAILED: Installation ended unsuccessfully (build directory: /tmp/example/Subread/2.0.1/GCC-8.5.0): build failed (first 300 chars):
+    Couldn't find file subread-2.0.1-source.tar.gz anywhere, and downloading it didn't work either...
+    Paths attempted (in order): ...
+    ```
+
+    In this case, the problem is that the easyconfig file does not specify
+    where the sources can be downloaded from. Not automatically at least,
+    but there is a helpful comment included:
+    ```python
+    # download from https://download.sourceforge.net/subread/subread-2.0.1-source.tar.gz
+    sources = ['subread-%(version)s-source.tar.gz']
+    ```
+
+    We can download the source tarball ourselves,
+    and move it to the location where EasyBuild expects to find it
+    (in the `sourcepath` directory):
+    ```
+    curl -OL https://download.sourceforge.net/subread/subread-2.0.1-source.tar.gz
+    mv subread-2.0.1-source.tar.gz $HOME/easybuild/sources/s/Subread/
+    ```
+
+    If downloading is problematic for some reason, the source tarball is also available
+    in `/easybuild/sources/s/Subread`.
+
+    Or, we can change the easyconfig file to specify the location where
+    the easyconfig file can be downloaded from:
+    ```python
+    source_urls = ['https://download.sourceforge.net/subread/']
+    sources = ['subread-%(version)s-source.tar.gz']
+    ```
+    Note that the `source_urls` value is a *list* of candidate download URLs,
+    *without* the filename of the source file itself.
+
+    This way, EasyBuild will download the source file when running `eb subread.eb`.
+
+    The source tarball is fairly large (23MB), so don't be alarmed if the download takes a little while.
+
+    ```shell
+    $ ls -lh $HOME/easybuild/sources/s/Subread
+    total 23M
+    -rw-rw-r-- 1 easybuild easybuild 23M Jun 13 17:42 subread-2.0.1-source.tar.gz
+    ```
+
+---
+
+***Exercise T.2**** - Toolchain*
+
+After fixing the problem with missing source file, try the installation again.
+
+What's wrong now? How can you fix it quickly?
+
+Take into account that we just want to get this software package installed,
+we don't care too much about details like the version of the dependencies or
+the *toolchain* here...
+
+
+??? success "(click to show solution)"
+
+    The installation fails because the easyconfig specifies that GCC 8.5.0
+    should be used as toolchain:
+    ```
+    $ eb subread.eb
+    ...
+    == FAILED: Installation ended unsuccessfully (build directory: /tmp/easybuild/Subread/2.0.1/GCC-8.5.0): build failed (first 300 chars):
+    No module found for toolchain: GCC/8.5.0 (took 1 sec)
+    ```
+
+    We don't have this GCC version installed, but we do have GCC 10.2.0:
```

Comment:
This should be 11.2.0 here and in the next lines for the setup that we have for ISC'22.